ext/drbbdup/drbbdup.c

```diff
@@ -478,15 +492,35 @@ drbbdup_duplicate_phase(void *drcontext, void *tag, instrlist_t *bb, bool for_tr
         drbbdup_set_up_copies(drcontext, bb, manager);
     }
 
-    dr_rwlock_write_unlock(rw_lock);
-
     /* If there's no dynamic handling, we do not need to store translations,
      * which saves memory (and is currently better supported in DR and drreg).
      */
     return manager->enable_dynamic_handling ? DR_EMIT_STORE_TRANSLATIONS
                                             : DR_EMIT_DEFAULT;
 }
 
+static dr_emit_flags_t
+drbbdup_duplicate_phase(void *drcontext, void *tag, instrlist_t *bb, bool for_trace,
+                        bool translating)
+{
+    dr_emit_flags_t emit_flags;
+
+    if (is_thread_private) {
+        drbbdup_per_thread *pt =
+            (drbbdup_per_thread *)drmgr_get_tls_field(drcontext, tls_idx);
+        emit_flags = drbbdup_do_duplication(&(pt->manager_table), drcontext, tag, bb,
```

Comment:
nit: Parentheses are superfluous around `(pt->manager_table)`, here and in several places below: `->` has higher precedence than `&` and the DR code base avoids parens in situations like this already so for consistency seems best to remove.

Reply:
Yep, I know! Personal taste from my part! :p

Reply:
Done.